File: src/main/resources/python/runtime/java/lang/Integer.py
class Integer(object):
    def __init__(self, value):
        self._value = value
    
    def equals__Object__boolean(self, other):
        # TODO: test this properly
        return isinstance(other, Integer) and other._value == self._value

    def __str__(self):
        return str(self._value)

    def __repr__(self):
        return str(self)

File: src/main/resources/csharp/runtime/java/lang/Integer.cs
namespace Couscous.java.lang {
    internal class Integer : Object {
        private readonly int _value;

        internal Integer(int value) {
            _value = value;
        }

        internal override bool equals__Object__boolean(Object other) {
            var integer = other as Integer;
            return integer != null && integer._value == _value;
        }
    }
}
